src/System.Management.Automation/engine/hostifaces/History.cs

```diff
@@ -1017,137 +1000,133 @@ public string Id
         /// </summary>
         protected override void EndProcessing()
         {
-            // Invoke-history can execute only one command. If multiple
-            // ids were provided, throw exception
-            if (_multipleIdProvided)
-            {
-                Exception ex =
-                    new ArgumentException
-                    (
-                        StringUtil.Format(HistoryStrings.InvokeHistoryMultipleCommandsError)
-                    );
-
-                ThrowTerminatingError
-                (
-                    new ErrorRecord
-                    (
-                        ex,
-                        "InvokeHistoryMultipleCommandsError",
-                        ErrorCategory.InvalidArgument,
-                        null
-                    )
-                );
-            }
-
             History history = ((LocalRunspace)Context.CurrentRunspace).History;
             Dbg.Assert(history != null, "History should be non null");
 
-            // Get the history entry to invoke
-            HistoryInfo entry = GetHistoryEntryToInvoke(history);
-
-            // Check if there is a loop in invoke-history
-            LocalPipeline pipeline = (LocalPipeline)((LocalRunspace)Context.CurrentRunspace).GetCurrentlyRunningPipeline();
-
-            if (pipeline.PresentInInvokeHistoryEntryList(entry) == false)
-            {
-                pipeline.AddToInvokeHistoryEntryList(entry);
-            }
-            else
+            // If the user didn't provide an Id, get the ID for the most
+            // recent entry
+            if (Id == null)
             {
-                Exception ex =
-                    new InvalidOperationException
-                    (
-                        StringUtil.Format(HistoryStrings.InvokeHistoryLoopDetected)
-                    );
-
-                ThrowTerminatingError
-                (
-                    new ErrorRecord
-                    (
-                        ex,
-                        "InvokeHistoryLoopDetected",
-                        ErrorCategory.InvalidOperation,
-                        null
-                    )
-                );
+                Id = new string[1]{history.GetEntries(0, 1, true)[0].Id.ToString()};
```

Comment:
What if History empty?

Reply:
Are we protected against that by L999 `Dbg.Assert(history != null, "History should be non null");`? Or is that inadequate here?

Reply:
That will only protect debug builds. If this is an expected result in release builds we'll need to write a more normal error instead (and probably just remove the assertion).

Or it could simply "do nothing" if there is no history, but even there you'll probably need to check for it and return early in that case.

Reply:
Simple test - run pwsh and run first command `Invoke-History`.

Reply:
![image](https://user-images.githubusercontent.com/13203965/85776714-d2d9cd80-b6d5-11ea-8fee-b32b5bc05e1d.png)

Yeah... that should probably be a better error message. I'll change that.

Reply:
See a message before the change.

Reply:
Sorry, @iSazonov, I'm not sure what exactly you mean there.

But, I've changed how the error is thrown and added a test to validate the type of error that gets thrown when there's no history.